### PR TITLE
fix incorrect inner_dim in AttnProcessor2_0

### PR DESCRIPTION
**Pull request by zhvng**
# What does this PR do?

Fix bug in AttnProcessor2_0 that arises when the size of the input to the attention layer != attention head dim * number of heads.  `inner_dim` should be based on the shape after `to_q`, `to_k` or `to_v`, as it is in the other attention processors.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by zhvng** [comment]
@patrickvonplaten

**Comment by patrickvonplaten** [comment]
Hey @zhvng,

Could you add a reproducible code snippet showing how the error can arise?

**Comment by zhvng** [comment]
of course! here's a snippet
```python
from diffusers.models.attention import Attention
from diffusers.models.attention_processor import AttnProcessor, AttnProcessor2_0, LoRAAttnProcessor2_0
import torch
import torch.nn as nn

# works
model = Attention(query_dim=512, heads=8, dim_head=64)
model.set_processor(AttnProcessor2_0())
model(torch.randn(1, 512, 512))

# doesn't work
model = Attention(query_dim=1024, heads=8, dim_head=64)
model.set_processor(AttnProcessor2_0())
model(torch.randn(1, 512, 1024))

# works with old attn processor 
model.set_processor(AttnProcessor())
model(torch.randn(1, 512, 1024))
```

The LoRA processors have the same issue but would require more than just a 1 line fix

**Comment by patrickvonplaten** [comment]
Cool that works! Feel free to also open a PR for LoRA :-)
